onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_execution_provider.cc

```diff
@@ -175,6 +175,14 @@ NnapiExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph_view
   result = utils::CreateSupportedPartitions(graph_viewer, is_node_supported, on_group_closed,
                                             gen_metadef_name, NNAPI, kNnapiExecutionProvider);
 
+  std::for_each(result.begin(), result.end(), [](auto& capability) {
+    if (capability && capability->sub_graph && capability->sub_graph->GetMetaDef()) {
+      if (capability->sub_graph->GetMetaDef()->inputs.empty()) {
```

Comment:
Please add a comment explaining why we're checking for empty inputs.

In general if you're adding code and the reason for its existence is not explicitly obvious, capture the reason in a comment so the next developer has that information. That will also help them learn what the code does faster and understand some of the implicit requirements.

In this case there are two things going on. a) a compute capability where all the inputs are coming from constant initializers; and b) NNAPI doesn't support a model with zero inputs. if you were new to the EP it would take a considerable amount of time to fully understand the reason for this check without that info.

Reply:
Thanks.

This lines of code does have special intention and need comments.